src/IKVM.Reflection/Impl/PortablePdbWriter.cs

```diff
@@ -0,0 +1,257 @@
+        {
+            var buffer = new ByteBuffer(16);
+            methodInfo.MethodSignature.WriteSig(moduleBuilder, buffer);
+            return builder.GetOrAddBlob(buffer.ToArray());
+        }
+
+        private static ParameterHandle GetParameterList(MetadataBuilder builder, MethodBase methodInfo)
+        {
+            ParameterHandle firstHandle = default;
+            int index = 1;
+            foreach (var parameter in methodInfo.GetParameters())
+            {
+                var handle = builder.AddParameter(System.Reflection.ParameterAttributes.None, builder.GetOrAddString(parameter.Name ?? ""), index);
+                if (firstHandle.IsNil)
+                {
+                    firstHandle = handle;
+                }
+                index++;
+            }
+            return firstHandle;
+        }
+
+        private void WriteScope(MetadataBuilder builder, MethodDefinitionHandle methodDef, Scope scope)
+        {
+            builder.AddLocalScope(
+                method: methodDef,
+                importScope: default,
+                variableList: MakeLocalVariables(builder, scope),
+                constantList: default,
+                startOffset: scope.startOffset,
+                length: scope.endOffset - scope.startOffset
+            );
+
+            // By spec doc:
+            //   > The table is required to be sorted first by Method in ascending order, then by StartOffset in ascending order, then by Length in descending order.
+            // https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#localscope-table-0x32
+            foreach (var child in scope.scopes.OrderBy(s => s.startOffset).ThenByDescending(s => s.endOffset - s.startOffset))
+            {
+                WriteScope(builder, methodDef, child);
+            }
+        }
+
+        private LocalVariableHandle MakeLocalVariables(MetadataBuilder builder, Scope scope)
+        {
+            LocalVariableHandle firstHandle = default;
+            foreach (var (name, local) in scope.locals.OrderBy(kv => kv.Value.addr1).Select(kv => (kv.Key, kv.Value)))
+            {
+                Debug.Assert(local.addrKind == System.Diagnostics.SymbolStore.SymAddressKind.ILOffset);
+                var handle = builder.AddLocalVariable(LocalVariableAttributes.None, local.addr1, builder.GetOrAddString(name));
+                if (firstHandle.IsNil)
+                {
+                    firstHandle = handle;
+                }
+            }
+            return firstHandle;
+        }
+
+        private BlobHandle MakeSequencePoints(MetadataBuilder builder, Method method)
+        {
+            // https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#sequence-points-blob
+            var writer = new BlobBuilder();
+
+            // header:
+            // LocalSignature
+            writer.WriteCompressedInteger(0); // Omit for now?
```

Comment:
Is this allowed? My implementation actually included a StandAloneSig.

Reply:
TBH I am not sure, but i don't see problems with vsdbg or netcoredbg.